test_architectural_fixes.py

```
@@ -0,0 +1,144 @@
+            lock1 = await manager1._get_state_lock()
+            lock2 = await manager1._get_state_lock()
+            lock3 = await manager2._get_state_lock()
+            
+            # All locks from same instance should be identical
+            assert lock1 is lock2, "Race condition: different locks created for same instance"
+            # Locks from different instances should be different
+            assert lock1 is not lock3, "Different instances should have different locks"
+            print("✅ process.py race condition fix verified")
+        
+        asyncio.run(test_get_state_lock())
+    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)